src/router.js

```diff
@@ -1,70 +1,84 @@
-export function Router(app, view) {
+export function Router(emit) {
   return {
     state: {
-      router: match(location.pathname)
+      router: {}
     },
     actions: {
       router: {
-        match: function(state, actions, data, emit) {
+        update: function(state, actions, router) {
           return {
-            router: emit("route", match(data))
+            router: router
           }
         },
-        go: function(state, actions, data) {
-          history.pushState({}, "", data)
-          actions.router.match(data.split("?")[0])
```

Comment:
So urls with query strings should certainly not break the router.. I guess we have two options:

1. Split the url at `?` take the first item in the split array and extract matched parts as `router.params` object. Ignore the second item in split array.
2. Split the url at `?` take the first item in the split array and extract matched parts as `router.params` object **and** take the second item in split array and extract parts as `router.query` object.